addons/Trail/trail_3d.gd

```diff
@@ -23,15 +23,19 @@ extends Node3D
 @export_enum("X", "Y", "Z") var axe: String = "Y"
 @export var show_wireframe: bool = false
 @export var wireframe_color: Color = Color(1, 1, 1, 1)
-@export var wire_line_width: float = 1.0 # (float, 0, 100, 0.1)
+@export var wire_line_width: float = 1.0  # (float, 0, 100, 0.1)
+@export var material: Material
 
 var points := []
 var color := Color(1, 1, 1, 1)
 var always_update = false
 
-var _target :Node3D
-var _wire_obj :ImmediateMesh = ImmediateMesh.new()
-var _wire_mat :StandardMaterial3D   = StandardMaterial3D.new()
+var _mesh_instance: MeshInstance3D = MeshInstance3D.new()
+var _immediate_mesh: ImmediateMesh = ImmediateMesh.new()
```

Comment:
It looks like in Godot 4, they changed it so that `ImmedateMesh` is a mesh type that you can assign to a `MeshInstance3D`.

I changed this one to create the `MeshInstance3D` when it loads and then assign the `ImmediateMesh` to that instance's `mesh` property.